core/src/doc/lives.rs

```diff
@@ -32,146 +32,169 @@ impl Document {
+			let lqopt = opt.new_with_perms(true).with_auth(Arc::from(auth));
+			// First of all, let's check to see if the WHERE
+			// clause of the LIVE query is matched by this
+			// document. If it is then we can continue.
+			let lqctx = lqctx.freeze();
+			match self.lq_check(stk, &lqctx, &lqopt, &lq, doc).await {
+				Err(Error::Ignore) => continue,
+				Err(e) => return Err(e),
+				Ok(_) => (),
+			}
+			// Secondly, let's check to see if any PERMISSIONS
+			// clause for this table allows this document to
+			// be viewed by the user who created this LIVE
+			// query. If it does, then we can continue.
+			match self.lq_allow(stk, &lqctx, &lqopt, &lq, doc).await {
+				Err(Error::Ignore) => continue,
+				Err(e) => return Err(e),
+				Ok(_) => (),
+			}
+			// Finally, let's check what type of statement
+			// caused this LIVE query to run, and send the
+			// relevant notification based on the statement.
+			if stm.is_delete() {
+				// Send a DELETE notification
+				if opt.id()? == lv.node.0 {
+					// Ensure futures are run
+					let lqopt: &Options = &lqopt.new_with_futures(true);
+					// Output the full document before any changes were applied
+					let mut result =
+						doc.doc.as_ref().compute(stk, &lqctx, lqopt, Some(doc)).await?;
+					// Remove metadata fields on output
+					result.del(stk, &lqctx, lqopt, &*META).await?;
+					let res = chn
+						.send(Notification {
 							id: lv.id,
 							action: Action::Delete,
 							record: Value::Thing(rid.as_ref().clone()),
-							result: {
-								// Ensure futures are run
-								let lqopt: &Options = &lqopt.new_with_futures(true);
-								// Output the full document before any changes were applied
-								let mut value =
-									doc.doc.as_ref().compute(stk, &lqctx, lqopt, Some(doc)).await?;
-								// Remove metadata fields on output
-								value.del(stk, &lqctx, lqopt, &*META).await?;
-								// Output result
-								value
-							},
+							result,
 						})
-						.await?;
-					} else {
-						// TODO: Send to message broker
+						.await;
+
+					if res.is_err() {
+						// channel was closed, that means a transaction probably failed.
+						// just return as nothing can be send.
+						return Ok(());
 					}
-				} else if self.is_new() {
-					// Send a CREATE notification
-					if opt.id()? == lv.node.0 {
-						chn.send(Notification {
+				} else {
+					// TODO: Send to message broker
+				}
+			} else if self.is_new() {
+				// Send a CREATE notification
+				if opt.id()? == lv.node.0 {
+					let res = chn
+						.send(Notification {
 							id: lv.id,
 							action: Action::Create,
 							record: Value::Thing(rid.as_ref().clone()),
 							result: self.pluck(stk, &lqctx, &lqopt, &lq).await?,
 						})
-						.await?;
-					} else {
-						// TODO: Send to message broker
+						.await;
+
+					if res.is_err() {
+						// channel was closed, that means a transaction probably failed.
+						// just return as nothing can be send.
```

Comment:
Shall we log (warning, or debug) to avoid having silent errors?

Reply:
It is not really an error. The error returned is due to the channel being closed and the channel is closed because the next notifications should be ignored because a transaction failed. The error being returned is expected behavior not something the user should be informed of.